x/mongo/driver/operation/find.go

```diff
@@ -146,7 +149,8 @@ func (f *Find) command(dst []byte, desc description.SelectedServer) ([]byte, err
 	if f.max != nil {
 		dst = bsoncore.AppendDocumentElement(dst, "max", f.max)
 	}
-	if f.maxTimeMS != nil {
+	// Only append specified maxTimeMS if timeout is not also specified.
+	if f.maxTimeMS != nil && f.timeout == nil {
 		dst = bsoncore.AppendInt64Element(dst, "maxTimeMS", *f.maxTimeMS)
 	}
```

Comment:
I'm concerned this logic for including `maxTimeMS` in the command document if `timeout` is not set, and then including it in the command doc in `createMsgWireMessage` if `timeout` is set, could be brittle if someone doesn't realize the blocks are related. Would it be better to pass `maxTimeMS` through to the `Operation` like we do with `timeout` and keep the `maxTimeMS` vs `timeout` logic all in one place?

E.g.
```go
func (f *Find) Execute(ctx context.Context) error {
	// ...
	return driver.Operation{
		// ...
		Timeout:   f.timeout,
		MaxTimeMS: f.maxTimeMS,
	}.Execute(ctx, nil)
}

func (op Operation) Execute(ctx context.Context, scratch []byte) error {
	// ...
	var maxTimeMS uint64
	if op.Timeout != nil {
		// ...
	} else if op.MaxTimeMS != nil {
		// ...
	}
	// ...
}
```

Reply:
Added a comment on GODRIVER-2342 with the recommendation. Sounds good to do that refactor later 👍